build/scripts/build-utils.ps1

```diff
@@ -0,0 +1,145 @@
+# Collection of powershell build utility functions that we use across our scripts.
+
+set-strictmode -version 2.0
+$ErrorActionPreference="Stop"
+
+# Declare a number of useful variables for other scripts to use
+[string]$repoDir = [IO.Path]::GetFullPath($(join-path $PSScriptRoot "..\.."))
+[string]$binariesDir = join-path $repoDir "Binaries"
+[string]$scriptDir = $PSScriptRoot
+
+# The intent of this script is to locate and return the path to the MSBuild directory that
+# we should use for bulid operations.  The preference order for MSBuild to use is as 
+# follows
+#
+#   1. MSBuild from an active VS command prompt
+#   2. MSBuild from a machine wide VS install
```

Comment:
@tmat 

My long term goal is to have the opposite model: default is xcopy, let developers opt into using a machine wide install.

But this is a fairly new endeavor and I'm imagining it will take a week or two to work out all of the kinks.  Hence I want to start conservatively.